dependencies.yaml

```diff
@@ -319,7 +319,7 @@ dependencies:
       - output_types: [conda, requirements, pyproject]
         packages:
           - &cython cython>=3.0.0
-          - treelite
+          - &treelite treelite==4.6.0
```

Comment:
Do we really need to pin to a specific version or could we make this a lower bound? I'm less concerned about building with a specific version, but we are re-using this as part of our runtime dependencies, too.